### PR TITLE
Add customize store colors

### LINKED ISSUE #39716: [Customize Your Store] Colors
## User Story

> As a new merchant, I would like a to fine tune the AI design, so I can add my personal touch.

## Design

* Mrk6SERPZ4KrFHSjM0a8TK-fi-2631%3A258874

## Acceptance Criteria

```
Scenario: Colors panel is opened
    Given the user opens the color panel
    Then available color palettes are shown
    And a section for the user to create their own is collapsed 

Scenario: User changes the color palette
    Given the user uploads an image
    Then the colors should be populated in the preview pane

Scenario: Save
    Given the user presses the save button
    Then the rendered preview should be saved permanently
```

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #39716.

Implement customize store color palette feature.

Figma: Mrk6SERPZ4KrFHSjM0a8TK-fi-2631%3A258874

https://github.com/woocommerce/woocommerce/assets/4344253/1f987b2a-818d-427b-a9cb-a3d4785f0ff0

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Use a block theme with a logo block, such as `FSE Empire` theme, or add a logo block to your theme using site-editor.
5. Go to `WooCommerce > Home`
6. Click on `Customize your store` task 
7. Click on `Assembler Hub` button
8. Click on "Change the color palette"
9. Confirm the UI matches the design
10. Select a color
11. Observe that the preview frame style should be changed.
12. Click on "or create your own"
13. Change "Background" color
14. Observe that the preview frame background color should be changed.
15. Click on "save" button
16. Changes should persist when reloading the page
17. Select a color without saving
18. Go back to main sidebar
19. Observe that changes should be discarded.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add customize store color palettes

#### Comment 

</details>

### CODE REVIEW

**Comment by ilyasfoo** [comment]
Hey, @chihsuan! A quick run in JN I've hit into white screen of death when viewing the CYS task, `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`. I've confirmed to be using WP 6.3 and FSE Empire theme. No other plugins installed.

image

**Comment by chihsuan** [comment]
> Hey, @chihsuan! A quick run in JN I've hit into white screen of death when viewing the CYS task, `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`. I've confirmed to be using WP 6.3 and FSE Empire theme. No other plugins installed.
> 
> image

Hey @ilyasfoo Sorry, I forgot to mention that we'll need to install Gutenberg plugin (>= 16.4) until we merge https://github.com/woocommerce/woocommerce/pull/40031.

**Review by psealock — COMMENTED** [review]
Upon clicking on "Assembler Hub", I get the following JS error:

>react-dom.min.js?ver=18:1 TypeError: (0 , L.dispatch)(...).__experimentalReapplyBlockTypeFilters is not a function

* JN site using WP 6.3.1
* `customize-store` turned on
* FSE Empire theme
* Gutenberg plugin installed and activated

Any ideas?

**Comment by ilyasfoo** [comment]
I'm getting the same error @psealock

**Comment by chihsuan** [comment]
Hey @psealock @ilyasfoo It's related to the latest Gutenberg, which was released a few hours ago. 😓 

The `__experimentalReapplyBlockTypeFilters` action [has been renamed to `reapplyBlockTypeFilters`](https://github.com/WordPress/gutenberg/pull/53807/files#diff-5cc3977e318367783d0292d63e984c66441bc21f29de1832de970f6b0aec291aL204-R57). Could you please use [16.5.1](https://github.com/WordPress/gutenberg/releases/tag/v16.5.1) for testing instead? cc @woocommerce/ghidorah 

I'll open another PR to fix the issue for the latest Gutenberg.

**Review by psealock — APPROVED** [review]
This is testing well for me with JN, nice work @chihsuan. I couldn't spot anything problematic with the code either. Thanks @chihsuan

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/style.scss [hunk 1]
```diff
@@ -427,3 +473,15 @@
...
+
+.woocommerce-customize-store_global-styles-variations_item {
+	border: 1px solid transparent;
+
+	&:hover,
+	&.is-active {
+		border-radius: 2px;
+		border: 1px solid #dcdcde;
+		background: #fff;
+		box-shadow: 0 0 0 4px  var(--wp-admin-theme-color), 0 0 0 2.5px #fff;
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/style.scss]: I think the selected border is a bit different compared to design


image


image

**Review by ilyasfoo — APPROVED** [review]
Tested well! Some minor comments but fine to be a follow-up. 🚢 

- One minor UX issue is, once a user has selected a color scheme, there's no way for them to revert it. Should we make it such that clicking on the active color scheme would deselect it and reset colors to default? cc @verofasulo if you have input
- Unrelated to this PR, but it seems the [styles](https://github.com/woocommerce/woocommerce/blob/7adc8f1a71e4dc6d9fcd357105f0c06227eab9a5/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-color-palette.tsx#L79) and editor link has double slash in it. Ex: `wp-admin//site-editor.php?path=%2Fwp_global_styles&canvas=edit`

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/style.scss (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/style.scss]: Good catch! 

It looks like the styles provided by figma need to be corrected, fixed the border style in 20218e76fd8cc0e24af2c45b23f439409f604824.

**Comment by chihsuan** [comment]
> * Unrelated to this PR, but it seems the [styles](https://github.com/woocommerce/woocommerce/blob/7adc8f1a71e4dc6d9fcd357105f0c06227eab9a5/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-color-palette.tsx#L79) and editor link has double slash in it. Ex: `wp-admin//site-editor.php?path=%2Fwp_global_styles&canvas=edit`

Thanks 😃  fixed in e6e6bb7ab841c9ac6dce9b8110131d723963d52e.

**Comment by chihsuan** [comment]
Going to merge this. Will address any other feedback in a follow-up.

**Pull request merged**